csharp/ql/src/AlertSuppression.ql

```diff
@@ -10,10 +10,12 @@ import csharp
 /**
  * An alert suppression comment.
  */
-class SuppressionComment extends SinglelineComment {
+class SuppressionComment extends CommentLine {
   string annotation;
 
   SuppressionComment() {
+    // Must be either `// ...` or `/* ... */` on a single line.
+    this.getRawText().regexpMatch("//.*|/\\*.*\\*/") and
```

Comment:
how about `exists(int line | this.getLocation().hasLocationInfo(_, line, _, line, _))` instead?

Reply:
That doesn't work because `CommentLine` are always 1 line long. `CommentBlock` is multiline however.

Reply:
Ah, right. Will `not exists(CommentLine other | other.getParent() = this.getParent() and other != this)` work?

Reply:
That doesn't work because the unit tests would (and did) fail. It combines adjacent comment lines irrespective of their type (single/multi-line) into a single `CommentBlock`. In particular,
```
/* ... */
/* ... */
```
would be considered to be a single `CommentBlock`.